Python/compile.c

```diff
@@ -125,18 +125,20 @@
         (c->c_flags->cf_flags & PyCF_ALLOW_TOP_LEVEL_AWAIT) \
         && (c->u->u_ste->ste_type == ModuleBlock))
 
-typedef struct location_ {
-    int lineno;
-    int end_lineno;
-    int col_offset;
-    int end_col_offset;
-} location;
+typedef PyCompilerSrcLocation location;
 
 #define LOCATION(LNO, END_LNO, COL, END_COL) \
     ((const location){(LNO), (END_LNO), (COL), (END_COL)})
 
 static location NO_LOCATION = {-1, -1, -1, -1};
 
+#define LOCATION_IS_GT(L1, L2) \
```

Comment:
Make this an inline function, please.
Also, maybe `compare_location(l1, l2)` for more generality?

Reply:
It's not a total order - how would this function work?

Reply:
Could be "lt, gt, or neither", and then you can't use it for equality. So we add an equality testing function too?

Reply:
I'll make a function just for gt for now - checking both directions would be less efficient (if it's not GT we don't care about the other distinction) and we don't have a use case for it now. We can always add a location_compare that does two GT calls if we need it.

Reply:
> It's not a total order - how would this function work?

Good point.